### DISCUSSION TITLE
I want to have a multilabel segmentation

### DISCUSSION THREAD

**Discussion by hannachoum** [open-mmlab/mmengine#1225] [General]
HI there, 

I have X-rays images that i need to segment. 
I have 35 classes, but each pixel in the image can be part of several classes, so i need to make multilabel segmentation.
In the Annotation folder, i have npz object which is a concatenation of 35 matrix (composed with 0,1). 
I need to modify the pipeline, in order to accept these objects. 
Can you help me ? 

Thank you for your help !

**Comment by HAOCHENYE** [comment]
Hi, sorry for my late reply. If you're using the [mmsegmentation](https://github.com/open-mmlab/mmsegmentation), I belive that you will get a faster and more accurate workaround there. If you're setting up your baseline directly based on MMEngine, the most simple way is to customize a Dataset that accepts your data format. The most important thing you need to do is to implement the `__getitem__` method and return the expected output.

**Reply by hannachoum** [reply]
HI, thank you, 
I already worked a lot on this case, but i need to rewrite all the transforms etc ... no ? Because i need them to accept my new data format.
do i also need to rewrite the loss of the model i'm using ?  
I don't understand how changing only the __getitem__ function can help me, 
Thank you,

**Comment by HAOCHENYE** [comment]
Without knowing the specifics of your code implementation, it is hard to provide precise advice. However, if you are facing an issue where the input data format required by transforms does not match the raw data loaded by the dataset and the input expected by `model.forward`, the simplest solution is to convert the raw data to the format required by the transforms. Additionally, ensure that the function signature of `model.f`orward aligns with the data returned by the transforms. For example, if the transforms return a dictionary with keys "inputs" and "data_samples," model.forward should accept these two parameters.
